include/compartment.h

```diff
@@ -0,0 +1,105 @@
+#ifndef _COMPARTMENT_H
+#define _COMPARTMENT_H
+
+#include <assert.h>
+#include <elf.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/auxv.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "cheriintrin.h"
+
+#include "manager.h"
+
+#define align_down(x, align)    __builtin_align_down(x, align)
+#define align_up(x, align)      __builtin_align_up(x, align)
+
+#define SEG_MAX_COUNT 20
+
+struct func_intercept;
+
+struct intercept_patch
+{
+    int* patch_addr;
```

Comment:
This is a minor nit, but it's better IMHO to follow the convention of `int *path_addr` because in C (surprisingly) `int* a, b` defines `int *a` and `int b`.

Reply:
Are you suggesting to move the pointer asterisk over to the variable? I understand the counter example, I just got into the habit of having the pointer near the type, since that makes sense to me in my head. I don't mind changing that for this example, if you think it would be best. I also generally avoid declaring multiple variables, unless it's a primitive type.

Reply:
Yeah, the asterisk should be right next to the variable name. It's one of C's quirks that doing this is a sign of good C programming ;)